extensions/binding/org.eclipse.smarthome.binding.digitalstrom/src/main/java/org/eclipse/smarthome/binding/digitalstrom/handler/DeviceHandler.java

```diff
@@ -300,120 +289,121 @@ private synchronized BridgeHandler getDssBridgeHandler() {
-                                    if (deviceStateUpdate.getValue() > 0) {
+                    if (currentChannel != null) {
+                        switch (deviceStateUpdate.getType()) {
+                            case DeviceStateUpdate.OUTPUT_DECREASE:
+                            case DeviceStateUpdate.OUTPUT_INCREASE:
+                            case DeviceStateUpdate.OUTPUT:
+                                if (currentChannel.contains(DsChannelTypeProvider.DIMMER)) {
+                                    if (deviceStateUpdate.getValueAsInteger() > 0) {
                                         updateState(new ChannelUID(getThing().getUID(), currentChannel),
-                                                new PercentType(fromValueToPercent(deviceStateUpdate.getValue(),
-                                                        device.getMaxOutputValue())));
+                                                new PercentType(
+                                                        fromValueToPercent(deviceStateUpdate.getValueAsInteger(),
+                                                                device.getMaxOutputValue())));
                                     } else {
                                         updateState(new ChannelUID(getThing().getUID(), currentChannel), OnOffType.OFF);
                                     }
-                                    break;
-                            }
-                            break;
-                        case DeviceStateUpdate.UPDATE_ON_OFF:
-                            switch (currentChannel) {
-                                case CHANNEL_ID_COMBINED_2_STAGE_SWITCH:
-                                case CHANNEL_ID_GENERAL_COMBINED_2_STAGE_SWITCH:
-                                    updateState(new ChannelUID(getThing().getUID(), currentChannel),
-                                            new StringType(convertStageValue((short) 2, device.getOutputValue())));
-                                    break;
-                                case CHANNEL_ID_COMBINED_3_STAGE_SWITCH:
-                                case CHANNEL_ID_GENERAL_COMBINED_3_STAGE_SWITCH:
-                                    updateState(new ChannelUID(getThing().getUID(), currentChannel),
-                                            new StringType(convertStageValue((short) 3, device.getOutputValue())));
-                                    break;
-                                default:
-                                    if (deviceStateUpdate.getValue() > 0) {
-                                        updateState(new ChannelUID(getThing().getUID(), currentChannel), OnOffType.ON);
+                                } else if (currentChannel.contains(DsChannelTypeProvider.STAGE)) {
+                                    if (currentChannel.contains(TWO_STAGE_SWITCH_IDENTICATOR)) {
+                                        updateState(new ChannelUID(getThing().getUID(), currentChannel),
```

Comment:
Could you replace all those occurrences in this class from channelUID to channelID. Here it would simply be:
```
updateState(currentChannel, new StringType(convertStageValue((short) 2, device.getOutputValue())));
```